### ISSUE TITLE
LT01: Allow spacing before alias expression to follow indentation size

### ISSUE THREAD

**Issue by RenanBasilio** [sqlfluff/sqlfluff#4928]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

To make our SQL more uniform, oftentimes in our projects any alias expressions are aligned to the nearest tab after at least one space, like so (consider a 4 space tab size):
```sql
select
    a.some_column       as x
    a.three_sp_column   as y
from ...
```
However, the current definition of `sqlfluff:layout:type:alias_expression` with `spacing_before = align` rule only accepts the following:
```sql
select
    a.some_column     as x
    a.three_sp_column as y
from ...
```
This request is to add a `spacing_before = align_indent` configuration which accepts additional whitespace so long as the alias expression is aligned with the configured `tab_space_size`.

### Use case

```sql
select
    a.some_column       as x
    a.three_sp_column   as y
from ...
```
Should pass rule LT01 when configured to do so.

### Dialect

Any

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
